sleap_roots/series.py

```diff
@@ -2,60 +2,102 @@
 
 import attrs
 import numpy as np
-from pathlib import Path
 import sleap_io as sio
-from typing import Optional, Tuple, List, Union
-
 import matplotlib
 import matplotlib.pyplot as plt
 import seaborn as sns
 
+from typing import Dict, Optional, Tuple, List, Union
+from attr import attrib
+from pathlib import Path
+
 
 @attrs.define
 class Series:
     """Data and predictions for a single image series.
 
     Attributes:
         h5_path: Path to the HDF5-formatted image series.
-        primary_labels: A `sio.Labels` corresponding to the primary root predictions.
-        lateral_labels: A `sio.Labels` corresponding to the lateral root predictions.
-        video: A `sio.Video` corresponding to the image series.
+        labels_dict: A dictionary where each key is a prediction name and the value is
+            a `sio.Labels` object corresponding to those predictions.
+        video: A `sio.Video` object corresponding to the image series.
+
+    Methods:
+        load: Class method to load a set of predictions.
+        series_name: Property that returns the series name derived from the HDF5 filename.
+        get_frame: Returns labeled frames for the given prediction name.
+        get_points: Returns points for the given prediction name.
+
+    Example:
+        >>> series = Series.load(h5_path="your_h5_path",
+        prediction_names=["primary_multi_day", "lateral_3_nodes"])
+        >>> primary_points =
+        series.get_points(frame_idx=0, pred_name="primary_multi_day")
     """
 
-    h5_path: Optional[str] = None
-    primary_labels: Optional[sio.Labels] = None
-    lateral_labels: Optional[sio.Labels] = None
-    video: Optional[sio.Video] = None
+    h5_path: Optional[str] = attrib(default=None)
+    labels_dict: Dict[str, sio.Labels] = attrib(factory=dict)
+    video: Optional[sio.Video] = attrib(default=None)
```

Comment:
Use `attrs.field` instead of `attr.attrib`.

Also, for optional fields with default of `None`, it's not necessary to have either one since `attrs` will auto-discover these fields.